docs/Setup/Chaincode-setup.md

```diff
@@ -1,70 +1,219 @@
 
-Multiple terminal windows inside the Vagrant development environment are required. One Vagrant terminal runs the validating peer; another Vagrant terminal runs the chaincode; the third Vagrant terminal runs the CLI or REST API commands to execute transactions. When running with security enabled, an additional fourth Vagrant terminal window is required to run the <b>Certificate Authority (CA)</b> server. Detailed instructions are provided in the sections below:
+These instructions apply to _developing_ chaincode in Go or Java. They do not apply to running in a production environment. However, if _developing_ chaincode in Java, please see the [Java chaincode setup](JAVAchaincode.md) instructions first, to be sure your environment is properly configured.
 
-* [Security Setup (optional)](#security-setup-optional)
-* [Vagrant Terminal 1 (validating peer)](#vagrant-terminal-1-validating-peer)
-* [Vagrant Terminal 2 (chaincode)](#vagrant-terminal-2-chaincode)
-* [Vagrant Terminal 3 (CLI or REST API)](#vagrant-terminal-3-cli-or-rest-api)
-    * [Chaincode deploy via CLI and REST](#chaincode-deploy-via-cli-and-rest)
-    * [Chaincode invoke via CLI and REST](#chaincode-invoke-via-cli-and-rest)
-    * [Chaincode query via CLI and REST](#chaincode-query-via-cli-and-rest)
-* [Removing temporary files when security is enabled](#removing-temporary-files-when-security-is-enabled)
+## Choices
 
-See the [logging control](../dev-setup/logging-control.md) reference for information on controlling
-logging output from the `peer` and chaincodes.
+Once again, you have the choice of using one of the following approaches:
 
-###Security Setup (optional)
-From your command line terminal, move to the `devenv` subdirectory of your workspace environment. Log into a Vagrant terminal by executing the following command:
+- [Option 1](#Option-1-Vagrant-development-environment) using the **Vagrant** [development environment](../dev-setup/devenv.md) that is used for developing the fabric itself
+- [Option 2](#Option-2-Docker-for-Mac-or-Windows) using Docker for Mac or Windows
+- [Option 3](#Option-3-Docker-Toolbox) using Docker toolbox
 
-    vagrant ssh
+Using options *2* and *3* allow you to avoid having to build from and keep current a clone of the fabric GitHub repos.
```

Comment:
3 suggestions: Using options 2 **and** 3 allow you to avoid having to **build from** and keep current a clone of the fabric GitHub **repos**.

How about:

Using option *2* or *3* will allow you to avoid having to build everything from scratch (along with the need to keep a clone of the fabric GitHub code current/up-to-date).

or (my preference, though I can't tell how this 'sounds'):

Using option 2 or 3 will preclude you from the need to build everything from scratch (and from keeping a clone of the fabric GitHub code current/up-to-date).